frontend/pages/home_page.js

```diff
@@ -1,33 +1,75 @@
 import React, { Component, PureComponent } from 'react';
 import { Redirect } from 'react-router';
 import appRoutes from 'src/constants/routes';
+import http from 'src/frontend/services/http';
 
 export default class HomePage extends Component {
   state = {
     loading: false,
     error: '',
     redirectTo: '',
+    products: '',
   }
 
   goToProductNewPage = () => {
     this.setState({ redirectTo: appRoutes.productNew });
   }
 
+  fetchProducts = () => {
+    http.get('/product').then(
+      data => {
+        this.setState({
+          products: data.products
+        })
+      }
+    ).catch(
+      e => {
+        console.log('error', e);
+      }
+    );
+  }
+
+  componentDidMount() {
+    this.fetchProducts();
+  }
+
   render() {
     const { error, redirectTo } = this.state;
+    const { products } = this.state;
+    console.log('products', products);
     if (redirectTo) return <Redirect push to={redirectTo} />;
-
-    return (
-      <div className='home-page'>
-        <header className='home-page__header'>
-          <div className='content'>
-            <button className='green-button' onClick={this.goToProductNewPage}>
-              <i className='fas fa-plus' /> New Product
-            </button>
+    if(!products) {
```

Comment:
we wanna render the home page regardless of whether there are products or not, but the div inside the home page should render either 'no products' or the products. So you can either conditionally render different stuff inline:

```
 return (
  <div className='home-page'>
   ...
  { !products ? <div>No Products</div> : getProductList() }
  </div>
)
```

or just make a render method that returns either or:

```
renderProducts() {
  const { products } = this.state;
  if (!products) return <div>No Products</div>;
  return ...
}
...
 return (
  <div className='home-page'>
   ...
  { renderProducts() }
  </div>
)
```